python/cuml/decomposition/tsvd_mg.pyx

```diff
@@ -0,0 +1,566 @@
+            tsvd_float.fit(gdf_float)
+
+            print(f'components: {tsvd_float.components_}')
+            print(f'explained variance: {tsvd_float.explained_variance_}')
+            print(f'explained variance ratio: {tsvd_float.explained_variance_ratio_}')
+            print(f'singular values: {tsvd_float.singular_values_}')
+
+            trans_gdf_float = tsvd_float.transform(gdf_float)
+            print(f'Transformed matrix: {trans_gdf_float}')
+
+            input_gdf_float = tsvd_float.inverse_transform(trans_gdf_float)
+            print(f'Input matrix: {input_gdf_float}')
+
+    Output:
+
+    .. code-block:: python
+
+            components:            0           1          2
+            0 0.58725953  0.57233137  0.5723314
+            1 0.80939883 -0.41525528 -0.4152552
+            explained variance:
+            0  55.33908
+            1 16.660923
+
+            explained variance ratio:
+            0  0.7685983
+            1 0.23140171
+
+            singular values:
+            0  7.439024
+            1 4.0817795
+
+            Transformed matrix:           0            1
+            0 5.1659107    -2.512643
+            1 3.4638448 -0.042223275                                                                                                                     2 4.0809603    3.2164836
+
+            Input matrix:           0         1         2
+            0       1.0  4.000001  4.000001
+            1 2.0000005 2.0000005 2.0000007
+            2  5.000001 0.9999999 1.0000004
+
+    For additional examples, see `the Truncated SVD  notebook <https://github.com/rapidsai/cuml/blob/master/python/notebooks/tsvd_demo.ipynb>`_. For additional documentation, see `scikitlearn's TruncatedSVD docs <http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.TruncatedSVD.html>`_.
+
+    """
+
+    def __init__(self, n_components=1, tol=1e-7, n_iter=15, random_state=None,
+                 algorithm='full'):
+        if algorithm in ['full', 'auto', 'jacobi']:
+            c_algorithm = self._get_algorithm_c_name(algorithm)
+        else:
+            msg = "algorithm {!r} is not supported"
+            raise TypeError(msg.format(algorithm))
+        self.params = TSVDparams(n_components, tol, n_iter, random_state,
+                                 c_algorithm)
+        self.components_ = None
+        self.explained_variance_ = None
+        self.explained_variance_ratio_ = None
+        self.singular_values_ = None
+        self.components_ptr = None
+        self.explained_variance_ptr = None
+        self.explained_variance_ratio_ptr = None
+        self.singular_values_ptr = None
+
+    def _get_algorithm_c_name(self, algorithm):
+        return {
+            'full': COV_EIG_DQ,
+            'auto': COV_EIG_DQ,
+            'jacobi': COV_EIG_JACOBI
+        }[algorithm]
+
+    def _initialize_arrays(self, n_components, n_rows, n_cols):
```

Comment:
You're using numpy to fill your arrays with zeros here which could be slow on large arrays. I know we have a `cudf.utils.cudautils.zeros` in cudf that you could use if this is a concern.

Reply:
Sweet, thats a great point!